clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -3111,6 +3289,47 @@ scheduler_tmpl_t<RecordType, ReaderType>::eof_or_idle(output_ordinal_t output)
         assert(options_.mapping != MAP_AS_PREVIOUSLY || outputs_[output].at_eof);
         return sched_type_t::STATUS_EOF;
     } else {
+        if (options_.mapping == MAP_TO_ANY_OUTPUT) {
```

Comment:
Would this class become any simpler if it were templatized based on options_.mapping? (not for this PR)

Reply:
I previously had related thoughts on subclassing to avoid the conditionals.  Filed #6831 and put XXX comments in several places in this file.